src/hooks/useNetworkCheck.tsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react'
+import ReactDOM from 'react-dom'
+import { ToastId } from 'react-toastify'
+import { toast } from 'toastify'
+
+import { useWalletConnection } from 'hooks/useWalletConnection'
+import useSafeState from 'hooks/useSafeState'
+
+import { Network } from 'types'
+
+const availableNetworks = CONFIG.exchangeContractConfig.config.map(({ networkId }) => networkId)
+
+const NetworkErrorMessage: React.FC = () => (
+  <div>
+    <div>Please switch to a supported network!</div>
+    <strong>{`${availableNetworks.map(
+      (id, idx, arr) => (idx === arr.length - 1 ? ' or ' : ' ') + Network[id],
+    )}`}</strong>
```

Comment:
Here you are using the fact that `array.toString()` basically does `array.join(',')`
Shouldn't do it like that
Better use:
```ts
availableNetworks.map((id, idx, arr) => (idx === arr.length -1 ? 'or ' : '') + Network[id]).join(', ')
```

and move it outside the component, no need to recalc on render.

Reply:
sure, can i ask why that's better though? genuinely curious

Reply:
(not the moving out, the .join)

Reply:
Specific `toString` behaviors are implementation details, that are generally not a good thing to depend on. They are there to make some data structures human readable in case of logging out using in string concatenation.
You wouldn't really on an object producing a nice string, you'll get `[object Object]` or specific class/constructor name. Functions will produce the whole function internals as a string. So I would not really on arrays too.
Nope there's nothing inherently wrong with stringifying arrays, just the conversation is very implicit, and can be overridden at times.